sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java

```diff
@@ -85,21 +162,50 @@
   protected final String fileNamingTemplate;
 
   /**
-   * Construct a FileBasedSink with the given base output filename and extension.
+   * Construct a FileBasedSink with the given base output filename and extension. A
+   * {@link OutputWrapperFactory} of type {@link CompressionType#UNCOMPRESSED} will be used.
    */
   public FileBasedSink(String baseOutputFilename, String extension) {
     this(baseOutputFilename, extension, ShardNameTemplate.INDEX_OF_MAX);
   }
 
   /**
+   * Construct a FileBasedSink with the given base output filename, extension, and
+   * {@link OutputWrapperFactory}.
+   */
+  public FileBasedSink(String baseOutputFilename, String extension,
+      OutputWrapperFactory outputWrapperFactory) {
+    this(baseOutputFilename, extension, ShardNameTemplate.INDEX_OF_MAX, outputWrapperFactory);
+  }
+
+  /**
    * Construct a FileBasedSink with the given base output filename, extension, and file naming
-   * template.
+   * template. A {@link OutputWrapperFactory} of type {@link CompressionType#UNCOMPRESSED}
+   * will be used.
    *
-   * <p>See {@link ShardNameTemplate} for a description of file naming templates.
+   * <p>
+   * See {@link ShardNameTemplate} for a description of file naming templates.
    */
   public FileBasedSink(String baseOutputFilename, String extension, String fileNamingTemplate) {
+    this(baseOutputFilename, extension, fileNamingTemplate, CompressionType.UNCOMPRESSED);
+  }
+
+  /**
+   * Construct a FileBasedSink with the given base output filename, extension, file naming template,
+   * and {@link OutputWrapperFactory}.
+   *
+   * <p>
+   * See {@link ShardNameTemplate} for a description of file naming templates.
+   */
+  public FileBasedSink(String baseOutputFilename, String extension, String fileNamingTemplate,
+      OutputWrapperFactory outputWrapperFactory) {
+    this.outputWrapperFactory = outputWrapperFactory;
     this.baseOutputFilename = baseOutputFilename;
-    this.extension = extension;
+    if (outputWrapperFactory.getFilenameSuffix() != null) {
```

Comment:
Is this allowed to be `null`? If so, please annotate the relevant methods and fields with `@Nullable`. If not, use `Strings.isNullOrEmpty` for a simpler check (that still gracefully handles bad data).